code/modules/art/paintings.dm

```diff
@@ -322,15 +326,16 @@
 
 /obj/item/canvas/proc/try_rename(mob/user)
 	if(painting_metadata.loaded_from_json) // No renaming old paintings
-		return
+		return TRUE
 	var/new_name = tgui_input_text(user, "What do you want to name the painting?", "Title Your Masterpiece")
-	if(new_name != painting_metadata.title && new_name && user.can_perform_action(src))
+	if(isnull(new_name))
+		return FALSE
+	if(new_name != painting_metadata.title && user.can_perform_action(src))
 		painting_metadata.title = new_name
 	var/sign_choice = tgui_alert(user, "Do you want to sign it or remain anonymous?", "Sign painting?", list("Yes", "No"))
 	if(sign_choice != "Yes")
 		painting_metadata.creator_name = "Anonymous"
```

Comment:
```suggestion
	switch(tgui_alert(user, "Do you want to sign it or remain anonymous?", "Sign painting?", list("Yes", "No", "Cancel"))
		if("Yes")
			return TRUE
		if("No")
			painting_metadata.creator_name = "Anonymous"
	
	return FALSE
```

A) I really like having both the "Cancel" option and respecting `null` here, because someone might screw up on an earlier tgui_input and want to cancel/close out of the prompt to try again.

B) For a proc like this, we want to default to returning `FALSE` for potential bullshit reasons, and only go `TRUE` when all of our criteria have been met and we are certain beyond any acts of God that this is what the user wanted.

C) You can keep the `var/sign_choice` line if you want, I just wrote it the way I typically would.

Reply:
The painting doesn't get finalized if you select "No". That's not intended behaviour,. If we select no it should set the creator_name and still return true